dataland-user-service/src/test/kotlin/org/dataland/datalanduserservice/service/PortfolioMonitoringServiceTest.kt

```diff
@@ -0,0 +1,114 @@
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.core.context.SecurityContextHolder
+import java.util.UUID
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class PortfolioMonitoringServiceTest {
+    private val mockPortfolioRepository = mock<PortfolioRepository>()
+    private val mockSecurityContext = mock<SecurityContext>()
+    private lateinit var portfolioMonitoringService: PortfolioMonitoringService
+
+    private val dummyUserId = "userId"
+    private val dummyPortfolioId = UUID.randomUUID().toString()
+    private val dummyCorrelationId = UUID.randomUUID().toString()
+
+    private val dummyPortfolio =
+        BasePortfolio(
+            portfolioId = dummyPortfolioId,
+            portfolioName = "Portfolio",
+            userId = dummyUserId,
+            creationTimestamp = System.currentTimeMillis(),
+            lastUpdateTimestamp = System.currentTimeMillis(),
+            companyIds = mutableSetOf("companyId"),
+            isMonitored = false,
+            startingMonitoringPeriod = null,
+            monitoredFrameworks = mutableSetOf("sfdr"),
+        )
+
+    @BeforeEach
+    fun setup() {
+        resetSecurityContext()
+        doAnswer { it.arguments[0] }.whenever(mockPortfolioRepository).save(any())
+        portfolioMonitoringService = PortfolioMonitoringService(mockPortfolioRepository)
+    }
+
+    private fun resetSecurityContext() {
+        val mockAuthentication =
+            AuthenticationMock.mockJwtAuthentication(
+                "username",
+                dummyUserId,
+                setOf(),
+            )
+        doReturn(mockAuthentication).whenever(mockSecurityContext).authentication
+        SecurityContextHolder.setContext(mockSecurityContext)
+    }
+
+    @Test
+    fun `verify that patchMonitoring throws PortfolioNotFoundApiException if portfolio not found`() {
+        val patch =
+            PortfolioMonitoringPatch(
+                isMonitored = true,
+                startingMonitoringPeriod = "2025-06-01",
+                monitoredFrameworks = mutableSetOf("sfdr"),
+            )
+
+        doReturn(null)
+            .whenever(mockPortfolioRepository)
+            .getPortfolioByUserIdAndPortfolioId(dummyUserId, UUID.fromString(dummyPortfolioId))
+
+        assertThrows<PortfolioNotFoundApiException> {
+            portfolioMonitoringService.patchMonitoring(dummyPortfolioId, patch, dummyCorrelationId)
+        }
+    }
+
+    @Test
+    fun `verify that non monitor values remain unchanged after patching monitoring`() {
+        val originalPortfolio =
+            dummyPortfolio.copy(
+                portfolioName = "Original Portfolio",
+                userId = dummyUserId,
+                companyIds = mutableSetOf("companyA", "companyB"),
+            )
+        val patch =
+            PortfolioMonitoringPatch(
+                isMonitored = true,
+                startingMonitoringPeriod = "2025-Q3",
```

Comment:
would that be a valid input?
```suggestion
                startingMonitoringPeriod = "2025",
```